### PR TITLE
agent: Add unit tests for sandbox.rs

### PR DESCRIPTION
**Pull request by ericho**
These are the unit tests for the sandbox struct. This is the summary
of the most important changes:

  - To test containers it was needed to create a `LinuxContainer` type
    and this requires root privileges. So, some tests now requires root
    user to be run.
  - There was a bug in the `unset_sandbox_storage` method. The return
    type was wrapped in a `Result` to avoid this problem.

Fixes: #50

Signed-off-by: Erich Cordoba <erich.cordoba.malibran@intel.com>

### CODE REVIEW

**Comment by ericho** [comment]
I have two questions regarding this code. Some of the tests are failing and i'm not sure if that is expected or I'm seeing some possible bugs.

File: src/agent/src/sandbox.rs [hunk 1]
```diff
@@ -266,3 +268,213 @@ fn online_memory(logger: &Logger) -> Result<()> {
...
+    #[allow(unused_assignments)]
+    fn unset_and_remove_sandbox_storage() {
+        skip_if_not_root!();
+
+        let logger = slog::Logger::root(slog::Discard, o!());
+        let mut s = Sandbox::new(&logger).unwrap();
+
+        // FIX: This test fails, not sure why yet.
+        // assert!(s.unset_and_remove_sandbox_storage("/tmp/testEphePath").is_err(),
+        //         "Should fail because sandbox storage doesn't exist");
```
Comment (ericho) [inline src/agent/src/sandbox.rs]: This call to `unset_and_remove_sandbox_storage` returns an `Ok`. But [in the golang](https://github.com/kata-containers/agent/blob/b634b06ca3717b732c0ec2398fb0b0a078f06b32/agent_test.go#L292-L295) version expects to fail.

File: src/agent/src/sandbox.rs [hunk 2]
```diff
@@ -266,3 +268,213 @@ fn online_memory(logger: &Logger) -> Result<()> {
...
+
+        // Add a new sandbox storage
+        assert_eq!(s.set_sandbox_storage(&storage_path), true);
+        // Use the existing sandbox storage
+        assert_eq!(s.set_sandbox_storage(&storage_path), false,
+                   "Expects false as the storage is not new.");
+
+        assert_eq!(s.unset_sandbox_storage(&storage_path), false,
+                   "Expects false as there is still a storage.");
+
```
Comment (ericho) [inline src/agent/src/sandbox.rs]: Here, `unset_sandbox_storage` is returning true, where as I understand it should return false.

Comment (ericho) [reply src/agent/src/sandbox.rs]: I changed the function to this (notice the `return true`: 
```rust
    pub fn unset_sandbox_storage(&mut self, path: &str) -> bool {
        match self.storages.get_mut(path) {
            None => return false,
            Some(count) => {
                *count -= 1;
                if *count < 1 {
                    self.storages.remove(path);
                    return true;
                }
                return false;
            }
        }
    }
```

I you agree I can change that function to this.

Comment (yyyeerbo) [reply src/agent/src/sandbox.rs]: I think you're right. Let @lifupan confirm it

Comment (lifupan) [reply src/agent/src/sandbox.rs]: Yeah, you are right. it's really a defect.

Comment (ericho) [reply src/agent/src/sandbox.rs]: Thanks @lifupan @yyyeerbo I'm fixing it!

**Comment by egernst** [comment]
@lifupan while I continue to work on becoming less rust-ignorant, can you PTAL?

@ericho -- nice PR, thanks for the help.

**Pull request merged**